app/controllers/FileValidationController.scala

```diff
@@ -59,8 +60,12 @@ class FileValidationController @Inject()(
               updatedAnswers <- Future.fromTry(UserAnswers(request.internalId).set(ValidXMLPage, fileName))
               updatedAnswersWithURL <- Future.fromTry(updatedAnswers.set(URLPage, downloadUrl))
               _              <- sessionRepository.set(updatedAnswersWithURL)
+              xml <- Future.successful(xmlValidationService.loadXML(downloadUrl))
             } yield {
-              Redirect(navigator.nextPage(ValidXMLPage, NormalMode, updatedAnswers))
+              (xml \ "DAC6Disclosures" \ "DisclosureImportInstruction").text match {
+                case "DAC6DEL" => Redirect(routes.DeleteDisclosureSummaryController.onPageLoad())
+                case _ => Redirect(navigator.nextPage(ValidXMLPage, NormalMode, updatedAnswers))
+              }
```

Comment:
we're loading the xml again to pull out one bit of data. Is there a way we can pass back some metadata along with the validation success?